File: src/main/java/com/tencent/bk/devops/atom/pojo/Result.java
package com.tencent.bk.devops.atom.pojo;

public class Result<T> {
    private int status; //状态码，0代表成功
    private String message; //描述信息
    private T  data; //数据对象

    public Result() {
    }

    public Result(T data) {
        this.status = 0;
        this.data = data;
    }

    public Result(int status, String message) {
        this.status = status;
        this.message = message;
        this.data = null;
    }

    public int getStatus() {
        return status;
    }

    public void setStatus(int status) {
        this.status = status;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public T getData() {
        return data;
    }

    public void setData(T data) {
        this.data = data;
    }

    public boolean isNotOk() {
        return this.status != 0;
    }

    public boolean isOk() {
        return this.status == 0;
    }
}

File: src/main/kotlin/com/tencent/bk/devops/plugin/pojo/Result.kt
package com.tencent.bk.devops.plugin.pojo

import com.fasterxml.jackson.annotation.JsonIgnore

data class Result<out T>(
    val status: Int = 0, // 状态码，0代表成功
    val message: String? = null, // 描述信息
    val data: T? = null
) { // 数据对象

    constructor(data: T) : this(0, null, data)
    constructor(status: Int, message: String?) : this(status, message, null)

    @JsonIgnore
    fun isOk(): Boolean {
        return status == 0
    }

    @JsonIgnore
    fun isNotOk(): Boolean {
        return status != 0
    }
}
